### ISSUE TITLE
DescriptionList - heading level and heading element invalid in examples

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#11393]
**Describe the problem**
This ranges from example code update to component update:

- For the Popover that triggers when clicking the term help text, the heading level is not correct for where it is rendered in the DOM. Either the heading should be an `h1` element, or the Popover needs to be appended in a way that the current `h6` is correct.
- For the Popover that triggers when clicking the term help text, axe complains about multiple banner landmarks. This is due to the Popover being appended to the document body and outside the `main` element; whenever a `header` is rendered outside of it, it becomes a banner landmark. Some ways to fix this could be:
  - update the element to a plain div
  - change where the Popover is appended (could also help resolve the first issue above)
  - since the Popover traps focus and has attributes similar to a Modal, all other content on the page gets set to aria-hidden="true", similar behavior to the Modal component. This might be an issue since we don't render a Backdrop for Popover like we do for Modal, though, so it may not be totally clear that content outside the Popover cannot be interacted with. Also, unlike Modal, you can click outside the Popover to close it, which may not make the `aria-modal="true"` attribute totally correct.

**How do you reproduce the problem?**
Provide steps to reproduce. A [codesandbox](https://codesandbox.io/s/serverless-cherry-q9t3f ) demonstrating the problem is appreciated.

**Expected behavior**
A clear and concise description of the expected behavior.

**Is this issue blocking you?**
List the workaround if there is one.

**Screenshots**
If applicable, add screenshots to help explain the issue.

**What is your environment?**

**What is your product and what release date are you targeting?**

**Any other information?**

**Comment by thatblindgeye** [comment]
https://github.com/patternfly/patternfly-react/issues/10852 has some related issues noted, but specifically for the Popover component. Potentially we could allow overriding the wrapper element for the header and footer components in Popover, so that when it's appended to doc body it can be plain divs, otherwise if it's appended within the `main` element the header and footer elements should be fine to use.
